grai-integrations/source-metabase/src/grai_source_metabase/loader.py

```diff
@@ -0,0 +1,295 @@
+import json
+from itertools import chain
+from typing import Optional, Callable, Dict, List
+
+import requests
+from pydantic import BaseSettings, SecretStr, validator
+from requests.exceptions import ConnectionError
+from retrying import retry
+
+from src.grai_source_metabase.models import Question, Table, NodeTypes, Edge
+
+_endpoint = "https://data.inv.tech/api"
+
+
+class MetabaseConfig(BaseSettings):
+    endpoint: str = _endpoint
+    username: SecretStr
+    password: SecretStr
+
+    @validator("endpoint")
```

Comment:
Since we are assuming this endpoint is along the `/api` path let's add some explicit validation 

```python
    @validator("endpoint")
    def validate_endpoint(cls, v):
        endpoint = v.rstrip("/") 
        if not endpoint.endswith('/api'):
            raise ValueError("The Metabase API endpoint must end with '/api'")
        return endpoint
```